sycl/source/detail/memory_manager.cpp

```diff
@@ -306,7 +306,7 @@ void MemoryManager::fill(SYCLMemObjT *SYCLMemObj, void *Mem, QueueImplPtr Queue,
 
   if (Dim == 1) {
     PI_CALL(RT::piEnqueueMemFill(
-        Queue->getHandleRef(), pi::pi_cast<RT::PiMem>(Mem), Pattern, PatternSize, Offset[0],
+        Queue->getHandleRef(), pi::pi_cast<RT::PiMem>(Mem), Pattern, PatternSize, Offset[0] * ElementSize,
```

Comment:
It seems this change is not related to buffers copying, please move to separate commit.